RDKEMW-7182 : Use Consolidated FFV Algo Repos in RDKV Voice SDK

garys2511 opened the pull request:
Reason for change: Update voice sdk to use consolidated far-field voice algorithms Gerrit repo.

Test Procedure: Refer to Jira ticket.

Priority: P2

Risks: Low

Signed-off-by: Gary Skrabutenas@comcast.com

garys2511 committed:
RDKEMW-7182 : Use Consolidated FFV Algo Repos in RDKV Voice SDK

Reason for change: Update voice sdk to use consolidated far-field voice
algorithms Gerrit repo.

Test Procedure: Refer to Jira ticket.

Priority: P2

Risks: Low

Signed-off-by: Gary Skrabutenas@comcast.com

dwolaver commented on CMakeLists.txt:
```diff
@@ -211,14 +211,17 @@ if(XRAUDIO_KWD_ENABLED)
+if(XRAUDIO_PPR_ENABLED)
+   target_compile_definitions(xr-voice-sdk PUBLIC XRAUDIO_PPR_ENABLED)
+endif()
+   target_link_libraries(xr-voice-sdk xr_dsp_algorithms)
```
Can this line 224 be moved to hal implementation instead? or continue using one of the existing link names 'xraudio-eos' or 'xraudio-dga' or choose a new one 'xraudio-blah'

garys2511 replied:
Thanks, Dave. I'll investigate.

garys2511 replied:
How about I change to a new name 'xraudio-ffv-algorithms' in keeping with existing 'xraudio-ffv-hal' library?

garys2511 committed:
RDKEMW-7182 : Use Consolidated FFV Algo Repos in RDKV Voice SDK

Reason for change: Update voice sdk to use consolidated far-field voice algorithms Gerrit repo.

Test Procedure: Refer to Jira ticket.

Priority: P2

Risks: Low

Signed-off-by: Gary_Skrabutenas@comcast.com

dwolaver commented on CMakeLists.txt:
```diff
@@ -209,14 +209,17 @@ if(XRAUDIO_KWD_ENABLED)
    target_link_libraries(xr-voice-sdk xraudio-kwd)
 endif()
 
+if(XRAUDIO_EOS_ENABLED OR XRAUDIO_DGA_ENABLED OR XRAUDIO_PPR_ENABLED) 
```
Is this outer if statement necessary?

garys2511 replied:
Unless I'm wrong, I think the if statement is necessary in one form or another. I'm trying to avoid loading the xraudio-ffv-algorithms more than once if one or more of those algorithms are enabled. Rather than an outer if statement, I can remove it and do this:
...
if(XRAUDIO_EOS_ENABLED OR XRAUDIO_DGA_ENABLED OR XRAUDIO_PPR_ENABLED) 
   target_link_libraries(xr-voice-sdk xraudio-ffv-algorithms)
endif()
...
Would this work?

dwolaver replied:
could the line below be put in the XRAUDIO_KWD_ENABLED block or is this library still used when its disabled?  This is all temporary so it's not a big deal either way.  Just move forward.

target_link_libraries(xr-voice-sdk xraudio_ffv_algorithms)

garys2511 replied:
According to Scott, this library could still be used, perhaps in a mid-field remote, without a KWD where detection might be done in the cloud.

dwolaver replied:
Then it would need to be open sourced (put in xr-voice-sdk repo) or another HAL interface created to accommodate it.

dwolaver reviewed (commented):
Just one question...

garys2511 committed:
RDKEMW-7182 : Use Consolidated FFV Algo Repos in RDKV Voice SDK

Reason for change: Update voice sdk to use consolidated far-field voice algorithms Gerrit repo.

Test Procedure: Refer to Jira ticket.

Priority: P2

Risks: Low

Signed-off-by: Gary_Skrabutenas@comcast.com

dwolaver reviewed (commented):
approved for QA test

dwolaver reviewed (approved):
Ready to go!
